### PR TITLE
Add posthog package

### PR DESCRIPTION
**Pull request by mands**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mands** [comment]
@conda-forge-admin, please ping team

requesting a review

File: recipes/posthog/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+requirements:
+  host:
+    - python >=3.7.0,<4.0.0
+    - pip
+  run:
+    - python >=3.7.0,<4.0.0
+    - requests >=2.7.0,<3.0.0
+    - six >=1.5.0,<2.0.0
+    - monotonic >=1.5.0,<2.0.0
+    - backoff >=1.6.0,<2.0.0
```
Comment (chrisburr) [inline recipes/posthog/meta.yaml:26]: I guess the exact pin is a bug upstream. Can you open a PR/issue upstream and apply a patch if you're going to loosen the requirement so you can keep the `pip check` test?

Comment (mands) [reply recipes/posthog/meta.yaml:26]: merged upstream in https://github.com/PostHog/posthog-python/pull/36 but not yet released to pypi - can we publish in the meantime as this is holding up an update to our main package?

**Comment by mands** [comment]
@conda-forge-admin, please ping team

requesting a review - have pushed a fix to upstream which in turn restores the `pip check`. There is a further update upstream which will remove the pin however would request to release this now in the meantime and will update further once that release is deployed as this is blocking our main package. Thanks!

**Comment by mands** [comment]
@conda-forge-admin, please ping team

as above - requesting a review as need is blocking a dependent package update

**Pull request merged**